app/src/org/commcare/activities/LoginActivity.java

```diff
@@ -392,13 +406,63 @@ public void dataPullCompleted() {
         ViewUtil.hideVirtualKeyboard(LoginActivity.this);
         CommCareApplication.notificationManager().clearNotifications(NOTIFICATION_MESSAGE_LOGIN);
 
+        handleConnectSignIn();
+
         Intent i = new Intent();
         i.putExtra(LOGIN_MODE, uiController.getLoginMode());
         i.putExtra(MANUAL_SWITCH_TO_PW_MODE, uiController.userManuallySwitchedToPasswordMode());
         setResult(RESULT_OK, i);
         finish();
     }
 
+    public void handleConnectSignIn() {
+        String appId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+        String username = uiController.getEnteredUsername();
+        String pass = uiController.getEnteredPasswordOrPin();
+        ConnectIDManager.rememberAppCredentials(appId, username, pass);
+    }
+
+    public void handleConnectButtonPress() {
+        selectedAppIndex = -1;
+        ConnectIDManager.handleConnectButtonPress(success -> {
+            updateConnectButton();
+
+            //NOTE: Enable this to launch new app picker activity
+//            if(success) {
+//                Intent i = new Intent(this, AppSelectActivity.class);
+//
+//                startActivityForResult(i, 1);
+//            }
+        });
+    }
+
+    private void updateConnectButton() {
+        uiController.setConnectButtonText(ConnectIDManager.getConnectButtonText(this));
+        uiController.setConnectButtonVisible(ConnectIDManager.shouldShowConnectButton());
+
+        uiController.updateConnectLoginState();
+    }
+
+    private void checkForSavedCredentials() {
+        if(ConnectIDManager.isUnlocked()) {
+            int selectorIndex = uiController.getSelectedAppIndex();
+            String selectedAppId = appIdDropdownList.size() > 0 ? appIdDropdownList.get(selectorIndex) : "";
+            String seatedAppId = CommCareApplication.instance().getCurrentApp().getUniqueId();
+            if (!uiController.isAppSelectorVisible() || selectedAppId.equals(seatedAppId)) {
+                AuthInfo.ProvidedAuth credentials = ConnectIDManager.getCredentialsForApp(seatedAppId, uiController.getEnteredUsername());
+                if (credentials != null) {
+                    //uiController.setUsername(credentials.username);
+                    uiController.setPasswordOrPin(credentials.password);
+
+                    //NOTE: This would auto-login, but we decided not to
+                    //initiateLoginAttempt(uiController.isRestoreSessionChecked());
+                } else {
+                    uiController.setPasswordOrPin("");

```

Comment:
checking again on https://github.com/dimagi/commcare-android/pull/2660#discussion_r1256267233

Reply:
This code is used to clear the pre-populated password, i.e. if the user chooses to lock or forget ConnectID. I could handle that somewhere more explicitly so it's not part of this method, is that worth doing?